client/src/app/site/pages/meetings/pages/participants/pages/participant-detail/pages/participant-detail-manage/components/participant-create-wizard/participant-create-wizard.component.ts

```diff
@@ -319,4 +321,32 @@ export class ParticipantCreateWizardComponent extends BaseMeetingComponent imple
             }
         }
     }
+
+    public get isLockedOutAndCanManage(): boolean {
+        const lockedOutHelper = this.personalInfoFormValue?.locked_out ?? this.user?.is_locked_out;
+        return lockedOutHelper && this.checkSelectedGroupsCanManage;
+    }
+
+    public get lockoutCheckboxDisabled(): boolean {
+        const notChanged = (this.personalInfoFormValue?.locked_out ?? null) === null;
+        const isLockedOut = this.user?.is_locked_out;
+        return notChanged && !isLockedOut && this.checkSelectedGroupsCanManage;
+    }
+
+    public updateLockout(): void {
+        const check = this.detailView.personalInfoForm.get(`locked_out`).disabled !== this.lockoutCheckboxDisabled;
+        if (check) {
+            if (this.lockoutCheckboxDisabled) {
+                this.detailView.personalInfoForm.get(`locked_out`).disable();
+            } else {
+                this.detailView.personalInfoForm.get(`locked_out`).enable();
+            }
+        }
+    }
+
+    private get checkSelectedGroupsCanManage(): boolean {
+        return (this.detailView.personalInfoForm.get(`group_ids`).value ?? [])
+            .map((id: Id): ViewGroup => this.groupRepo.getViewModel(id))
+            .some(group => group.hasPermission(Permission.userCanManage));
+    }
```

Comment:
Properties are normally specified above the constructor. But I would prefer changing this to a normal method.

```suggestion
    private checkSelectedGroupsCanManage(): boolean {
        return (this.detailView.personalInfoForm.get(`group_ids`).value ?? [])
            .map((id: Id): ViewGroup => this.groupRepo.getViewModel(id))
            .some(group => group.hasPermission(Permission.userCanManage));
    }
```

Reply:
I change these two places to be functions instead of properties.